mender-update/cli/actions.cpp

```diff
@@ -28,9 +37,62 @@ namespace conf = mender::common::conf;
 namespace database = mender::common::key_value_database;
 namespace events = mender::common::events;
 namespace log = mender::common::log;
-
-namespace daemon = mender::update::daemon;
+namespace error = mender::common::error;
+namespace kv_db = mender::common::key_value_database;
+namespace path = mender::common::path;
 namespace standalone = mender::update::standalone;
+namespace daemon = mender::update::daemon;
+
+error::Error MaybeInstallBootstrapArtifact(context::MenderContext &main_context) {
+	const string bootstrap_artifact_path {
+		main_context.GetConfig().paths.GetBootstrapArtifactFile()};
+	// Check if the DB is populated - then install conditionally
+	auto &db = main_context.GetMenderStoreDB();
+	auto exp_key = db.Read(main_context.artifact_name_key);
+	if (exp_key) {
+		// Key exists. Do nothing
+		return error::NoError;
+	}
+	error::Error err = exp_key.error();
+	if (err != error::NoError
+		and err.code != kv_db::MakeError(kv_db::KeyError, "Key Not found").code) {
+		return err;
+	}
+
+	// Key does not exist, install the bootstrap artifact if it exists
+	if (!path::FileExists(bootstrap_artifact_path)) {
+		log::Debug("No Bootstrap Artifact found at: " + bootstrap_artifact_path);
+		error::Error err =
+			db.Write(main_context.artifact_name_key, common::ByteVectorFromString("unknown"));
+		if (err != error::NoError) {
+			return err;
+		}
+		return error::NoError;
+	}
+	log::Info("Installing the bootstrap Artifact");
+	const bool verify_artifact_signature = false;
+	auto result =
+		standalone::Install(main_context, bootstrap_artifact_path, verify_artifact_signature);
+
+	if (result.err != error::NoError) {
+		error::Error err =
+			db.Write(main_context.artifact_name_key, common::ByteVectorFromString("unknown"));
+		if (err != error::NoError) {
+			return err;
+		}
+		bool deleted = path::FileDelete(bootstrap_artifact_path);
+		if (!deleted) {
+			return context::MakeError(
+				context::ValueError, "Failed to delete the bootstrap Artifact");
```

Comment:
`ValueError`? If you change `FileDelete` to return an error, then you can just return it as is (possibly using `WithContext`). 

So maybe:
```
return result.err.FollowedBy(err.WithContext("Failed to delete the bootstrap Artifact"));
```